### PR TITLE
Fix offscreen rendering for Chrome 70

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Change List
- Use the latest `AnimationLoop` API - `AnimationLoopProxy.createWorker` now takes a single `AnimationLoop` instance as argument
- Support mouse events

#### Core Example Test Results
- x cubemap  (accesses `document`)
- ✓ fragment
- ✓ instancing
- ✓ mandelbrot
- x picking  (`loadTextures` accesses `Image`)
- ✓ shadowmap
- ✓ transform
- ✓ transform-feedback

### CODE REVIEW

File: examples/core/cubemap/app.js [hunk 1]
```diff
@@ -221,6 +221,6 @@ function drawTexture({ctx, sign, axis, size}) {
 export default animationLoop;
 
 /* global window */
-if (!window.website) {
+if (typeof window !== 'undefined' && !window.website) {
```
Comment (1chandu) [inline examples/core/cubemap/app.js]: Super nit: Create a util method given this is used in multiple places?

File: modules/core/src/core/animation-loop-proxy.js [hunk 2]
```diff
@@ -106,12 +139,49 @@ export default class AnimationLoopProxy {
...
+      case 'removeEventListener':
+        this.canvas.removeEventListener(evt.data.type, this._onEvent);
+        break;
+
+      default:
+    }
+  }
+
+  _onEvent(evt) {
+    const devicePixelRatio = this.props.useDevicePixels ? (window.devicePixelRatio || 1) : 1;
```
Comment (1chandu) [inline modules/core/src/core/animation-loop-proxy.js]: Is this code common to on-screen and off-screen? If it is only for offscreen, isn't `useDevicePixels` always false?

File: modules/core/src/core/animation-loop.js [hunk 3]
```diff
@@ -213,7 +218,7 @@ export default class AnimationLoop {
     this.onRender(this.animationProps);
     // end callback
 
-    if (this.offScreen) {
+    if (this.offScreen && this.gl.commit) {
```
Comment (1chandu) [inline modules/core/src/core/animation-loop.js]: Who sets this `this.gl.commit`, may be add a comment?

File: examples/core/cubemap/app.js (hunk 1 above)
Comment (Pessimistress) [reply examples/core/cubemap/app.js]: This is per example as they need to work independently.

File: modules/core/src/core/animation-loop-proxy.js (hunk 2 above)
Comment (Pessimistress) [reply modules/core/src/core/animation-loop-proxy.js]: `useDevicePixels` is false on the worker thread but defined by user on the main thread.

File: modules/core/src/core/animation-loop.js (hunk 3 above)
Comment (Pessimistress) [reply modules/core/src/core/animation-loop.js]: https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/commit

Calling `gl.commit` is required in FF but not in Chrome 70.

**Pull request merged**